packages/snaps-utils/src/manifest/validation.ts

```diff
@@ -141,7 +143,62 @@ export const SnapIdsStruct = refine(
 
 export type SnapIds = Infer<typeof SnapIdsStruct>;
 
-export const ChainIdsStruct = array(ChainIdStruct);
+export const ChainIdsStruct = refine(
+  size(array(ChainIdStruct), 1, Infinity),
+  'Chain IDs',
+  (value) => {
+    if (Array.isArray(value) && value.every((val) => isChainId(val))) {
```

Comment:
Does ChainIdStruct not already validate this?

Reply:
removed this part